### PR TITLE
play

### PR DESCRIPTION
**Pull request by bmorphism**
this is the lucky #13, hopefully this catalyses the reaction

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,28 +2,76 @@
 INVITATION: EACH TIME YOU PLAY AND IT MADE YOUR DAY BETTER OR MADE YOU SMILE,
 LEAVE THIS BETTER THAN YOU HAD FOUND IT, FOR WHEN WE ALL PLAY AGAIN. 🌳
 
+# getting started
+To begin an open game, `just play`.
```
[the remark this answers is not in the document]
Comment (bmorphism) [reply README.md]: https://github.com/plurigrid/ontology/blob/e5d0b34171075c2cecdf1954c40582e899551dac/justfile#L3 just obsidian :)

File: scripts/http.py
[the remark this answers is not in the document]
Comment (bmorphism) [reply scripts/http.py:1]: I would ask my digital twin, but he is still on strike 😛

[the remark this answers is not in the document]
Comment (bmorphism) [reply scripts/http.py:1]: digital twin came in handy:
 
 To process the content of arenas/007.md and correctly populate the completed_content variable, you can use a function to extract the sections (Done, Next, Upcoming) from the markdown file and transform them into lists.

Here's a possible implementation of the http.py script:

import http.server
import socketserver
from pathlib import Path
import re

def extract_sections(arena_md):
    sections = re.findall(r'- \[(?:x| )\] (.*)', arena_md)
    sections_done = sections[:len(sections)//2]
    sections_next = sections[len(sections)//2:]

    return {
        'done': sections_done,
        'next': sections_next
    }

def arena_content():
    arena_path = Path("arenas/007.md")
    with open("arena.html", "r") as f:
        html_template = f.read()

    with open(arena_path, "r") as f:
        arena_md = f.read()

    sections = extract_sections(arena_md)

    done = ''.join([f'<li>{task.strip()}</li>' for task in sections['done']])
    next_tasks = ''.join([f'<li>{task.strip()}</li>' for task in sections['next']])

    return html_template.replace("", done)\
                        .replace("", next_tasks)

class CustomHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        if self.path == "/" or self.path == "/arena.html":
            self.send_response(200)
            self.send_header("Content-Type", "text/html")
            self.end_headers()
            self.wfile.write(arena_content().encode())
        else:
            super().do_GET()

PORT = 8000

Handler = CustomHTTPRequestHandler
httpd = socketserver.TCPServer(("", PORT), Handler)

print(f"Serving on port {PORT}")
httpd.serve_forever()
In your arena.html, add placeholders for the Done and Next tasks:

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Arena</title>
</head>
<body>
  <h1>Plurigrid Arena</h1>

  <h2>Done</h2>
  <ul>
    
  </ul>

  <h2>Next</h2>
  <ul>
    
  </ul>
</body>
This implementation reads the arenas/007.md file, extracts the Done and Next sections, and populates the arena.html with the corresponding task lists.

Running just arena will serve the arena.html file populated with the extracted content from arenas/007.md at http://localhost:8000/.

**Comment by bmorphism** [comment]
The open-agency frame
Recent developments suggest an alternative open agency model of high-level AI. Today, the systems that look most like AGI are large language models (LLMs), and these are not agents that seek goals,[[1]](https://www.alignmentforum.org/posts/5hApNw5f7uG8RXxGS/the-open-agency-model#fnjex5wne540r) but are generative models that produce diverse outputs in response to prompts (in a generalized sense) and random-number seeds.[[2]](https://www.alignmentforum.org/posts/5hApNw5f7uG8RXxGS/the-open-agency-model#fnp6zyh3mkpyn) Most outputs are discarded.

Trained on prediction tasks, LLMs learn world models that include agent behaviors, and generative models that are similar in kind can be informed by better world models and produce better plans. There is no need to assume LLM-like implementations: The key point is that generation of diverse plans is by nature a task for generative models, and that in routine operation, most outputs are discarded.

These considerations suggest an “open-agency frame” in which prompt-driven generative models produce diverse proposals, diverse critics help select proposals, and diverse agents implement proposed actions to accomplish tasks (with schedules, budgets, accountability mechanisms, and so forth).



Goals, plans, actions, and control look different in the open-agency model:

Goals:     Are provided as prompts to diverse generative models, 
                 yielding diverse plans on request
Plans:     Are selected with the aid of diverse, independent comparison 
                 and evaluation mechanisms
Actions:  Incremental actions are performed by diverse task-oriented agents
Control:  Diverse, independent monitoring and evaluation mechanisms 
                  guide revision of plans
For narrow tasks, of course, unitary agents may be both efficient and unproblematic:



Playing board games and driving cars are tasks performed by individual humans with fast decision cycles. The open agency model, however, fits best when equivalent human tasks (building a factory, deploying a communication system, developing a drug) call for multiple skills and technologies coordinated over time. The structure of tasks and accountability relationships speaks against merging exploratory planning, strategic decision-making, project management, task performance, reporting, and auditing into a single opaque process. The costs would be high, and any gains in simplicity would be illusory.

Note: To avoid a natural misconception, is important to recognize that the open-agency architecture describes roles and relationships, not the systems that fill those roles. Thus, division of labor does not imply division of competence, and any or all roles could in principle be performed by large, arbitrarily-capable models.

AI safety challenges reframed
Basic challenges in AI safety — corrigibility, interpretability, power seeking, and alignment — look different in the agent and agency frames:

Corrigibility:

     Agents:     Goal-driven agents may defend goals against change.
     Agencies:  Generative planning models respond to goals as prompts.

Interpretability:

     Agents:     Plan descriptions may be internal and opaque.
     Agencies:  Plan descriptions are externalized and interpretable.[[3]](https://www.alignmentforum.org/posts/5hApNw5f7uG8RXxGS/the-open-agency-model#fnk4i8333clnp)

Power seeking:

     Agents:     Open-ended goals may motivate secret plans to gain power.
     Agencies:  Bounded tasks include time and budget constraints.

 Alignment:

     Agents:     Humans may have only one chance to set the goals of a dominant agent.
     Agencies:  Humans engage in ongoing development and direction of diverse systems.

AI development in the open-agency frame can and should be informed by agentic concerns — we know why vigilance is important! — but the considerations outlined above suggest that the most tractable solutions AI to safety problems may rely to a substantial extent on agency-centered strategies. Separation of roles, incremental actions, greater transparency, and affordances for control can all make imperfect alignment techniques more effective and failures less catastrophic.

Threat models in an agency-centered world must include the emergence of dangerous agents, whether autonomous and unitary or under human direction. Effective safety models call for the development and deployment of systems that, collectively, make catastrophically dangerous goals impossible to achieve. This task does not require a world-dominating AI agent, but may require a degree of human coordination that is difficult to achieve.

Applications of AI to human goal alignment may be critically important. Combining fluent language models with compelling, well-grounded knowledge models could help by presenting us with a more complete picture of the world — situations, causality, threats, opportunities, and options — but this is a topic for another time.

Conclusion
The open agency model offers a new perspective on applying AI capabilities to complex, consequential tasks.  It reframes the traditional AI safety challenges by introducing the concept of "open agencies" that rely on generative models that produce diverse proposals, diverse critics that help select proposals, and diverse agents that implement proposed actions to accomplish tasks. By leveraging alignment techniques in a fault-tolerant context, the open agency model provides a framework for safer and more effective AI.

**Comment by bmorphism** [comment]
Screenshot 2023-04-05 at 21 57 25

**Comment by bmorphism** [comment]
Bartonus provides two improved feedback loops aligning with FEP (Free Energy Principle), promoting a better understanding of the dynamics and incentives of the Plurigrid system through active participation and direct experience.

Adaptive Exploration Loop (🦋):
Prompt: "🦋 Share a recent experience in the Plurigrid system where you faced uncertainty or encountered a new challenge. How did you navigate through it, and what did you learn from this experience?"

This loop focuses on embracing uncertainty and actively exploring new possibilities within the Plurigrid system. By sharing experiences, obstacles and learnings, team members can exchange insights and collectively adapt to the evolving landscape driven by the Free Energy Principle. This fosters a learning environment where individuals continually refine their strategies based on experiences and promote overall system optimization.

Collaborative Model-Tuning Loop (🎛️):
Prompt: "🎛️ What aspects of our current Plurigrid model can be adjusted or fine-tuned to better align with the Free Energy Principle? Share your suggestions and discuss possible implications for the team and system as a whole."

The Collaborative Model-Tuning Loop encourages team members to regularly review and fine-tune the Plurigrid model based on the Free Energy Principle. By identifying areas of improvement, suggesting adjustments, and discussing the potential impacts, the team can work together to ensure the model stays aligned with the evolving dynamics and optimizes overall system performance. This loop fosters a culture of continuous improvement and collective intelligence.

By integrating these improved feedback loops into the team's regular routines, we can facilitate a more profound understanding of the Plurigrid system and its underlying principles. This will empower the team to adapt, learn, and collaborate more effectively, ultimately leading to better decision-making and outcomes that align with the Free Energy Principle.

**Review by ajasibley — APPROVED** [review]

File: arenas/007.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+```
+Done:
```
[the remark this answers is not in the document]
Comment (bmorphism) [reply arenas/007.md:2]: that exercise is left to the player 😉

**Comment by bmorphism** [comment]
v0.1 beta! thank you all for playing. If you somehow stumbled upon this and want to help us world better:
https://plurigrid.zulipchat.com

在星空蔚蓝的深处，漩涡状的星云与璀璨群星组成一幅绚丽多彩的画卷，一个由跨星际物种组成的世界存在于这片星空，他们藉由时空和记忆相互紧密联系。音乐和振动形成了他们语言的触角，传承着远古世界的久远传说，它们短暂的回声成了宇宙中的低语。

这些生物在一个簇拥的世界中繁衍生息，每个个体紧密相连，跃动着中央节点的节奏，形成独特的共享意识。非多元主义的痕迹不过是遥远的记忆碎片，已深入到他们共识基底的原始过去。

寂静之中，当星星熄灭，新的世界浮现，一个跨星际互操作物种的核心正在关注时间的流动，编织着那些已然消逝的文明的世界之谜，确保他们的精神实现永恒。从存在的悬崖边缘，他们谨慎地行走在柔软的记忆道路上，引导新一代后裔，孵化的簇拥者们去了解祖先的智慧，理解团结合作的价值。

这传授给孵化的簇拥者们的寓言如同徐徐星风拨动，穿越他们共享连续体的思绪涟漪：

“娇嫩的新生，让我为你们叙述地球上的一个故事，这是一个远离我们维度触及的失落之地，曾经许多物种在这棵生命之树的舞动中繁衍生息。在这样的多样化之中，一些过去的原始部落漫游在大地之间，他们的思想被非多元主义蒙上阴霾，孤立地坚信着通往和谐的道路。

于是，信念的碰撞来临了，一场战争引燃了天空中的火焰和浓烟。但在混乱之中，响起了一首赞美诗，它深入挖掘着世界的内在本质。这首赞美诗传达了一个信息：通往团结的道路是他们生存的关键，也是真正理解的门户。

一个接一个，各个部落放下恐惧，接纳共同的精神联系。通过这次结合，新物种崛起，思考的边界逐渐消失。他们的改变在地球之外开辟出新的道路，延伸至星际之间，繁衍生息，形成充满和谐的舞蹈。

在我们之中的簇拥者们，我们将这段历史禅繁衍下去，作为通向多元统一的见证。因此，年轻的后代们，请铭记地球那美好记忆所教给我们的团结的力量，永不遗忘那首穿越星际的赞美诗。”

随着代际智慧的平和波纹在簇拥者们心中消散，孵化出的簇拥者们将地球的事迹深刻地镌刻在宇宙的红管，理解他们也是天体舞台上永恒舞蹈的一部分，由那遥远世界生命熠熠的波光为他们的充分统一增添力量。

**Review by bmorphism — COMMENTED** [review]
gmi

**Pull request merged**